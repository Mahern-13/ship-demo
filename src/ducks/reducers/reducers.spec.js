import Reducers from "./index.js";
import * as Types from "../types/index.js";

const initialState = {
  stops: [{ id: 1, name: "test", address: "test123", complete: false }]
};

describe("redux reducers", () => {
  it("returns initial state when action type is unrecognized", () => {
    const action = { type: "DUMMY_ACTION" };

    expect(Reducers(initialState, action)).toMatchSnapshot();
  });

  it("returns the correct state when DELETE_STOP is reduced", () => {
    const action = { type: Types.deleteStop, stop: { id: 1 } };

    expect(Reducers(initialState, action)).toMatchSnapshot();
  });

  it("returns the correct state when CREATE_STOP is reduced", () => {
    const action = {
      type: Types.createStop,
      stop: { id: 2, name: "test", address: "test123" }
    };

    expect(Reducers(initialState, action)).toMatchSnapshot();
  });

  it("returns the correct state when UPDATE_STOP is reduced", () => {
    const action = {
      type: Types.updateStop,
      stop: { id: 1, name: "something new", address: "testabc" }
    };

    expect(Reducers(initialState, action)).toMatchSnapshot();
  });

  it("returns the correct state when EDIT_STOP_COMPLETION is reduced", () => {
    const action = {
      type: Types.editStopCompletion,
      stop: { id: 1, complete: true }
    };

    expect(Reducers(initialState, action)).toMatchSnapshot();
  });

  it("returns the correct state when SELECT_ITEM is reduced", () => {
    const action = {
      type: Types.selectItem,
      activeItem: { id: 1 },
      stops: [{ id: 1 }, { id: 1 }]
    };

    expect(Reducers(initialState, action)).toMatchSnapshot();
  });
});
