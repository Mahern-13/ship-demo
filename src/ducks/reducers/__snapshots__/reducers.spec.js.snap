// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux reducers returns initial state when action type is unrecognized 1`] = `
Object {
  "alert": null,
  "didAddRoute": false,
  "editingStepId": null,
  "error": null,
  "isCreatingStop": false,
  "routes": Array [
    1,
  ],
  "stops": Array [
    Object {
      "address": "test123",
      "completed": false,
      "id": 1,
      "name": "test",
      "verified": true,
    },
  ],
}
`;

exports[`redux reducers returns the correct state when COMPLETION_OF_STOP is reduced 1`] = `
Object {
  "alert": null,
  "didAddRoute": false,
  "editingStepId": null,
  "error": null,
  "isCreatingStop": false,
  "routes": Array [
    1,
  ],
  "stops": Object {
    "0": Object {
      "address": "test123",
      "completed": false,
      "id": 1,
      "name": "test",
      "verified": true,
    },
    "1": Object {
      "address": "test123",
      "completed": true,
      "id": 1,
      "name": "test",
      "verified": false,
    },
  },
}
`;

exports[`redux reducers returns the correct state when DELETE_STOP is reduced 1`] = `
Object {
  "alert": null,
  "didAddRoute": false,
  "editingStepId": null,
  "error": null,
  "isCreatingStop": false,
  "routes": Array [
    1,
  ],
  "stops": Object {
    "0": Object {
      "address": "test123",
      "completed": false,
      "id": 1,
      "name": "test",
      "verified": true,
    },
  },
}
`;

exports[`redux reducers returns the correct state when UPDATE_STOP is reduced 1`] = `
Object {
  "alert": false,
  "didAddRoute": false,
  "editingStepId": null,
  "error": false,
  "isCreatingStop": false,
  "routes": Array [
    1,
  ],
  "stops": Object {
    "0": Object {
      "address": "test123",
      "completed": false,
      "id": 1,
      "name": "test",
      "verified": true,
    },
    "1": Object {
      "address": "testabc",
      "completed": false,
      "id": 1,
      "name": "something new",
      "verified": false,
    },
  },
}
`;
